Source/JavaScriptCore/runtime/VM.cpp

```diff
@@ -1373,6 +1373,35 @@ void VM::drainMicrotasks()
     finalizeSynchronousJSExecution();
 }
 
+void VM::drainMicrotasks(JSGlobalObject* globalObject)
+{
+    if (m_drainMicrotaskDelayScopeCount) [[unlikely]]
+        return;
+
+    if (executionForbidden()) [[unlikely]]
+        m_defaultMicrotaskQueue.clear();
+    else {
+        do {
+            m_defaultMicrotaskQueue.performMicrotaskCheckpoint(*this,
+                [&](QueuedTask& task) ALWAYS_INLINE_LAMBDA {
+                    if (task.globalObject() != globalObject)
+                        return QueuedTask::Result::Suspended;
+                    if (RefPtr dispatcher = task.dispatcher())
+                        return dispatcher->run(task);
+
+                    runJSMicrotask(task.globalObject(), task.identifier(), task.job(), task.arguments());
+                    return QueuedTask::Result::Executed;
+                });
+            if (hasPendingTerminationException()) [[unlikely]]
+                return;
+            didExhaustMicrotaskQueue();
+            if (hasPendingTerminationException()) [[unlikely]]
+                return;
+        } while (!m_defaultMicrotaskQueue.isEmpty(globalObject));
```

Comment:
This is a fairly expensive operation (see `isEmpty(JSGlobalObject*)` implementation) because we iterate of the entire queue to see if any tasks are for the object we are handling and we do that over and over until we clear the queue, I think this is acceptable for now but should probably be addressed later.